python/ray/_private/gcs_utils.py

```diff
@@ -302,6 +319,114 @@ def pin_runtime_env_uri(self, uri: str, expiration_s: int) -> None:
             )
 
 
+class GcsAioClient:
+    def __init__(
+        self,
+        channel: Optional[GcsChannel] = None,
+        address: Optional[str] = None,
+    ):
+        if channel is None:
+            assert isinstance(address, str)
+            channel = GcsChannel(gcs_address=address, aio=True)
+        assert isinstance(channel, GcsChannel)
+        assert channel._aio is True
+        self._channel = channel
+        self._connect()
+
+    def _connect(self):
+        self._channel.connect()
+        self._kv_stub = gcs_service_pb2_grpc.InternalKVGcsServiceStub(
+            self._channel.channel()
+        )
+
+    async def internal_kv_get(
+        self, key: bytes, namespace: Optional[bytes], timeout: Optional[float] = None
+    ) -> Awaitable[bytes]:
+        logger.debug(f"internal_kv_get {key} {namespace}")
+        req = gcs_service_pb2.InternalKVGetRequest(namespace=namespace, key=key)
+        reply = await self._kv_stub.InternalKVGet(req, timeout=timeout)
+        if reply.status.code == GcsCode.OK:
+            return reply.value
+        elif reply.status.code == GcsCode.NotFound:
+            return None
```

Comment:
This is not an Awaitable[bytes]. The caller has to check that the return value is not None before calling await on the return object. It is better to return an Awaitable here that indicates the None situation (e.g., empty bytes)

Reply:
I think once the funciton id async function, it doesn't matter here. It'll be converted to await automatically:

```
(base) yic@ip-172-31-58-40:~ $ cat x.py
import asyncio

async def f():
    return None

async def main():
    v = await f()
    print("v=", v)

asyncio.run(main())
(base) yic@ip-172-31-58-40:~ $ python x.py
v= None
(base) yic@ip-172-31-58-40:~ $
```